src/conn.rs

```diff
@@ -157,6 +154,12 @@ impl Store {
             sqlite: connection,
         })
     }
+
+    pub fn fast_forward_user_partition(&mut self, new_head: Entid) -> Result<()> {
```

Comment:
What I expect to see here is essentially a compare-and-set: if my local partition map is _this_, then expand it to _this one_.

Doing that work requires writing to the DB, so it shouldn't be on `Store`, it should be on `InProgress`; if the DB write fails the metadata will be discarded.

And you want to advance all parts — typically `user` and `tx` — at the same time, atomically, right? Remember that the data itself will refer to tx IDs — it's not enough to only work with the user partition!